runtime/main/src/main/java/org/qbicc/runtime/main/VMHelpers.java

```diff
@@ -69,13 +71,14 @@ private static boolean isAssignableTo(Object value, type_id toTypeId, int toDime
         // putchar(':');
         // printInt(toDimensions);
         // putchar('\n');
-        if (toDimensions == 0) {
+        int intDim = toDimensions.intValue();
+        if (intDim == 0) {
             return isAssignableToLeaf(valueTypeId, toTypeId);
         } else if (ObjectModel.is_reference_array(valueTypeId)) {
-            int valueDims = ObjectModel.dimensions_of(value);
+            uint8_t valueDims = ObjectModel.dimensions_of(value);
             if (valueDims == toDimensions) {
```

Comment:
I'd be tempted to keep VMHelpers and ObjectModel viewing array dimensions as an int as they are both "runtime" code and it makes it easier to reason about what they mean.

I highlighted this code as from a pure java view of the world, `==` isn't the right operation here as we're comparing two `uint8_t` objects but it's OK in qbicc because we lower them to the same kind of data.  If we still treat them as `int` in this code, we allow the underlying model to store them as smaller values (good for footprint) and this code to be more obviously correct.

It's only at the boundary - in the ObjectModel intrinsics - we'd need to translate between the two views.

@dmlloyd Any concerns with that approach?

Reply:
I don't think it makes much difference.  The APIs are designed specifically to be able to do this for cases like this, so it's more of a stylistic question at that point.

Reply:
I don't feel strongly about this though my preference is to work with Java's primitive types here.  Either way is OK.